modules/renter/upload.go

```diff
@@ -74,6 +70,15 @@ func (r *Renter) Upload(up modules.FileUploadParams) error {
 			up.PieceSize = smallPieceSize
 		}
 	}
+	// calculate end height
+	// TODO: using 0 as a special value is kind of hacky. Should probably have
+	// an explicit "renew" field.
+	var endHeight types.BlockHeight
+	if up.Duration == 0 {
+		endHeight = 0
+	} else {
+		endHeight = r.cs.Height() + up.Duration
+	}
```

Comment:
why do you use '0' as a special value? Why do you need a 'renew' field?

Please resolve this TODO in this PR